test/Dialect/Torch/maximize-value-semantics.mlir

```diff
@@ -59,6 +59,28 @@ func @mutation_followed_by_view_like_ops(%value_t: !torch.vtensor, %overwriter:
   return %value_result : !torch.vtensor
 }
 
+// CHECK-LABEL:   func @mutation_of_view_like_op_result(
+// CHECK-SAME:                                             %[[VALUE_T:.*]]: !torch.vtensor, %[[OVERWRITER:.*]]: !torch.vtensor, %[[INT_LIST:.*]]: !torch.list<!torch.int>) -> !torch.vtensor {
+// CHECK:           return %[[OVERWRITER]] : !torch.vtensor
+func @mutation_of_view_like_op_result(%value_t: !torch.vtensor, %overwriter: !torch.vtensor, %int_list: !torch.list<!torch.int>) -> !torch.vtensor {
+  %t = torch.copy.to_tensor %value_t : !torch.tensor
+  %view = torch.aten.view %t, %int_list : !torch.tensor, !torch.list<!torch.int> -> !torch.tensor
+  torch.overwrite.tensor.contents %overwriter overwrites %view : !torch.vtensor, !torch.tensor
+  %result = torch.copy.to_vtensor %view : !torch.vtensor
+  return %result : !torch.vtensor
+}
+
+// CHECK-LABEL:   func @value_tensor_used_after_copy_was_mutated(
+// CHECK-SAME:                                                       %[[VALUE_T:.*]]: !torch.vtensor,
+// CHECK-SAME:                                                       %[[OVERWRITER:.*]]: !torch.vtensor) -> (!torch.vtensor, !torch.vtensor) {
+// CHECK:           return %[[VALUE_T]], %[[OVERWRITER]] : !torch.vtensor, !torch.vtensor
+func @value_tensor_used_after_copy_was_mutated(%value_t: !torch.vtensor, %overwriter: !torch.vtensor) -> (!torch.vtensor, !torch.vtensor) {
```

Comment:
I added this test to ensure that the `overwrite` op is always rewritten before the `copy.to_tensor` op. If the copy op is rewritten first, then the overwritten tensor will be replaced with `%value_t`, which means that when the overwrite is rewritten, the return of `%value_t` will be replaced with `%overwriter`, which is not correct.